src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -9095,7 +9095,14 @@ and TcEventItemThen (cenv: cenv) overallTy env tpenv mItem mExprAndItem objDetai
     let (SigOfFunctionForDelegate(delInvokeMeth, delArgTys, _, _)) = GetSigOfFunctionForDelegate cenv.infoReader delTy mItem ad
     let objArgs = Option.toList (Option.map fst objDetails)
     MethInfoChecks g cenv.amap true None objArgs env.eAccessRights mItem delInvokeMeth
-
+    
+    let ilEventDefCustomAttrs =
+        match einfo with
+        | ILEvent(ILEventInfo(_, ilEventDef))-> ilEventDef.CustomAttrs
```

Comment:
We shouldn't probably match directly on `EventInfo`, but instead have a method on it, like it's done in `MethInfo`:
https://github.com/dotnet/fsharp/blob/30b3a8fa53bf48da2b79492cddd98b4d219336a3/src/Compiler/Checking/infos.fs#L1147-L1150

Here
https://github.com/dotnet/fsharp/blob/30b3a8fa53bf48da2b79492cddd98b4d219336a3/src/Compiler/Checking/infos.fs#L2097

Reply:
Updated following your suggestion . Thanks for the review :)